rfcs/modal-component.md

```diff
@@ -0,0 +1,154 @@
+
+* `$isOpen: boolean`
+* `$size: string|number`
+* `$role: string`
+* `$closable: boolean`
+* `$theme: theme`
+
+More props likely TBA as development continues
+
+### Usage
+
+Basic usage:
+
+```javascript
+import * as React from 'react';
+import {Modal, SIZE} from 'baseui/modal';
+
+class App extends React.Component {
+  state = {
+    open: true
+  };
+  render() {
+    return (
+      <div>
+        <Button onClick={() => this.setState({open: true})}>Open</Button>
+        <Modal
+          isOpen={this.state.open}
+          close={() => this.setState({open: false})}
+        >
+          {({close}) => (
+            <>
+              <ModalTitle>Whatsup!</ModalTitle>
+              <ModalBody>This is a modal</ModalBody>
+              <ModalFooter>
+                <Button type=”tertiary” onClick={close}>Close</Button>
+                <Button type=”primary” onClick={this.submit}>Let’s Go</Button>
+              </ModalFooter>
+            </>
+          )}
+        </Modal>
+      </div>
+
+    );
+  }
+}
+```
+
+### Key Design Decisions
+
+* Modal content will not render server side. This is because portals don't exist server-side, and realistically your page shouldn't be loading with a modal open.
+
+### Accesibility
+
+Good read: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html
+
+* Upon opening, first focusable item in modal will receive focus (form input, footer button, etc)
+* Dialog element will `aria-modal="true"`
+* Explicitly expose props for role, aria-labeledby, aria-describedby, id, to encourage good a11y practices
+* `Tab` key will move between focusable items (form inputs, footer buttons, etc). According to best practices, tabbing past the last focusable item should loop back to the first focusable item.
+* `Escape` key will close the modal
+* Click on backdrop (anywhere outside dialog) will hide modal
+* Background will not be scrollable while modal is open (position: fixed)
+* Focus should ideally be transitioned back to last focused element upon closing
+
+### Possible Future Props
+
+The following props could be useful in the future but are low priority for V1
+
+* `onBackdropClick: function` - Callback handler for when backdrop is clicked specifically
+* `backdrop: boolean` - Set to false to not show a transparent backdrop
+* `backdropCloseable: boolean` - Set to false if click on backdrop should not close modal
+* `onEscapeKey: function` - Callback handler for when escape key is hit
+* `escakeKeyCloseable: boolean` - Set to false to disable escape key closing modal
+* `onOpenComplete: function` - handler after opening (after animation, etc)
+* `onCloseComplete: function` - handler after closing (after animation, etc)
+* `autofocus: boolean` - Set to false if modal shouldn't autofocus on its content
+* `enforceFocus: boolean` - Set to false if we should force focus to stay in modal
+* `usePortal: boolean` - Set to false to not use a portal, and just mount as direct child
```

Comment:
if mount node is provided, I think it's enough condition to use Portal or not

Reply:
Good point